src/workflows-app/SubmissionConfig.test.js

```diff
@@ -419,30 +428,1484 @@ describe('BaseSubmissionConfig with workflowsAppStore', () => {
+      status: AppProxyUrlStatus.Ready,
+      state: 'https://lz-abc/wds-abc-c07807929cd1/',
+    });
+  });
+
+  it('should call Leo to get proxy url for WDS if its not ready', async () => {
+    // ** ARRANGE **
+    const mockRunSetResponse = jest.fn(() => Promise.resolve(runSetResponse));
+    const mockMethodsResponse = jest.fn(() => Promise.resolve(methodsResponse));
+    const mockSearchResponse = jest.fn(() => Promise.resolve(searchResponses.FOO));
+    const mockTypesResponse = jest.fn(() => Promise.resolve(typesResponse));
+    const mockWdlResponse = jest.fn(() => Promise.resolve('mock wdl response'));
+    const mockLeoResponse = jest.fn(() => Promise.resolve(mockAzureApps));
+
+    Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            getForMethod: mockRunSetResponse,
+          },
+          methods: {
+            getById: mockMethodsResponse,
+          },
+        },
+        WorkspaceData: {
+          queryRecords: mockSearchResponse,
+          describeAllRecordTypes: mockTypesResponse,
+        },
+        WorkflowScript: {
+          get: mockWdlResponse,
+        },
+        Apps: {
+          listAppsV2: mockLeoResponse,
+        },
+      };
+    });
+
+    workflowsAppStore.set({
+      workspaceId: 'abc-c07807929cd1',
+      wdsProxyUrlState: { status: AppProxyUrlStatus.None, state: '' },
+      cbasProxyUrlState: { status: AppProxyUrlStatus.Ready, state: 'https://lz-abc/terra-app-abc/cbas' },
+      cromwellProxyUrlState: { status: AppProxyUrlStatus.Ready, state: 'https://lz-abc/terra-app-abc/cromwell' },
     });
 
+    // ** ACT **
+    await act(async () =>
+      render(
+        h(BaseSubmissionConfig, {
+          methodId: '123',
+          name: 'test-azure-ws-name',
+          namespace: 'test-azure-ws-namespace',
+          workspace: mockAzureWorkspace,
+        })
+      )
+    );
+
+    // ** ASSERT **
+    expect(mockRunSetResponse).toHaveBeenCalledTimes(1);
+    expect(mockTypesResponse).toHaveBeenCalledTimes(1);
+    expect(mockMethodsResponse).toHaveBeenCalledTimes(1);
+    expect(mockSearchResponse).toHaveBeenCalledTimes(1);
+    expect(mockWdlResponse).toHaveBeenCalledTimes(1);
+    // Leo is called since proxy urls for WDS is not available yet
+    expect(mockLeoResponse).toHaveBeenCalledTimes(1);
+
     // verify that 'wdsProxyUrlState' in store was updated
     expect(workflowsAppStore.get().wdsProxyUrlState).toStrictEqual({
       status: AppProxyUrlStatus.Ready,
       state: 'https://lz-abc/wds-abc-c07807929cd1/',
     });
   });
 });
+
+describe('Initial state', () => {
+  beforeAll(() => {
+    Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 1000 });
+    Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 800 });
+  });
+
+  beforeEach(() => {
+    getConfig.mockReturnValue({ wdsUrlRoot, cbasUrlRoot });
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
```

Comment:
Mocks are cleared automatically via global Jest configuration (see the `jest` section in package.json).
https://jestjs.io/docs/configuration#clearmocks-boolean